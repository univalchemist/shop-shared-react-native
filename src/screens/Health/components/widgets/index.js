export { default as CallToActionLink } from './CallToActionLink';
export { default as CardContainer } from './CardContainer';
export { default as CardLoader } from './CardLoader';
export { default as ErrorCard } from './ErrorCard';
export { default as TipCard } from './TipCard';
