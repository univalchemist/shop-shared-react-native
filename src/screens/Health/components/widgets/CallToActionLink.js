import React from 'react';
import { Text, Flex, Image, Divider } from '@wrappers/components';
import styled from 'styled-components/native';
import theme from '@theme';

const StyledText = styled(Text)`
  color: ${theme.colors.fonts.blackLink};
`;

const StyledImage = styled(Image)`
  width: 16px;
  height: 16px;
  align-self: center;
  margin-left: 10;
`;

export const CallToActionLink = ({ text, icon }) => {
  return (
    <Flex height={54} flexDirection="column">
      <Divider full={true} />
      <Flex
        flexDirection="row"
        paddingLeft={24}
        paddingRight={24}
        paddingTop={16}
      >
        <StyledText>{text}</StyledText>
        {icon && <StyledImage source={icon} />}
      </Flex>
    </Flex>
  );
};
