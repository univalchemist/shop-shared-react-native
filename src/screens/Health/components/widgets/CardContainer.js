import styled from 'styled-components/native';
import { TouchableHighlight } from 'react-native';
import { Box } from '@wrappers/components';

let applyStyle = comp => {
  return styled(comp)`
    background-color: ${props => props.theme.colors.white};
    height: ${props => (props.height ? props.height : 252.5)};
    border-top-color: ${props => props.color};
    border-top-width: 4;
    border-radius: 4;
    padding-top: 28;
    padding-bottom: 40;
    padding-left: 24;
    padding-right: 24;
    display: flex;
    shadow-offset: 0px 2px;
    shadow-opacity: 0.05px;
    shadow-radius: 3;
    shadow-color: #000;
  `;
};

export const CardContainer = applyStyle(Box);

export const TouchableCardContainer = applyStyle(TouchableHighlight);
