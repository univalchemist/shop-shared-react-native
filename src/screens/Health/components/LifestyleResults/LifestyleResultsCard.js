import React from 'react';
import { Box, Flex, Image, Text } from '@wrappers/components';
import { useTheme, useIntl } from '@wrappers/core/hooks';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import { LIFESTYLE_TIPS_MODAL } from '@routes';
import { TouchableHighlight } from 'react-native';
import { CallToActionLink } from '@screens/Health/components/widgets/CallToActionLink';
import { categories, logAction } from '@store/analytics/trackingActions';

export const Container = styled(TouchableHighlight)`
  background-color: ${props => props.theme.colors.white};
  border-top-color: ${props => props.color};
  flex-grow: 1;
  border-top-width: 4;
  border-radius: 4;
  padding-top: 28;
  display: flex;
  shadow-offset: 0px 2px;
  shadow-opacity: 0.05px;
  shadow-radius: 3;
  shadow-color: #000;
`;

export const Score = styled(Text)`
  color: ${props => props.color};
  font-size: ${props => props.theme.fontSizes[5]};
  font-weight: ${props => props.theme.fontWeights.bolder};
  line-height: ${props => props.theme.lineHeights[5]};
  letter-spacing: -1.5;
  margin-top: 6;
  margin-right: 30;
`;

const Title = styled(Text)`
  color: ${props => props.theme.colors.gray[0]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  letter-spacing: 0.3;
`;

const Status = styled(Text)`
  color: ${props => props.theme.colors.gray[0]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  font-weight: ${props => props.theme.fontWeights.bold};
  letter-spacing: 0.3;
`;

const Description = styled(Text)`
  padding-top: 32;
  color: ${props => props.theme.colors.gray[8]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 24;
  letter-spacing: 0.3;
  padding-bottom: 32;
`;

const LifestyleResultsCard = ({
  color,
  score,
  title,
  description,
  status,
  iconImage,
  tips,
  textColor,
  navigation,
}) => {
  const theme = useTheme();
  const intl = useIntl();

  return (
    <Container
      accessible={true}
      accessibilityRole="button"
      underlayColor={theme.colors.touchableOverlayColor}
      theme={theme}
      color={color}
      onPress={() => {
        logAction({
          category: categories.LIFESTYLE_OVERVIEW,
          action: `View result ${tips.category}`,
        });
        navigation.navigate(LIFESTYLE_TIPS_MODAL, { tips });
      }}
    >
      <>
        <Flex flexGrow={1} flexDirection={'row'} px={24}>
          <Box alignSelf={'center'} marginRight={30}>
            {iconImage ? (
              <Image
                accessible={true}
                accessibilityLabel={intl.formatMessage({
                  id: 'health.accessibilityLabel.icon',
                })}
                source={iconImage}
              />
            ) : (
              <Score
                accessible={true}
                accessibilityLabel={score.toFixed(1)}
                color={textColor}
              >
                {score.toFixed(1)}
              </Score>
            )}
          </Box>
          <Flex flexGrow={1} flexDirection={'column'}>
            <Title accessible={true} accessibilityLabel={title} flexGrow={1}>
              {title}
            </Title>
            <Status accessible={true} accessibilityLabel={status} flexGrow={1}>
              {status}
            </Status>
          </Flex>
          <Description
            accessible={true}
            accessibilityLabel={description}
            flexGrow={1}
          >
            {description}
          </Description>
        </Flex>
        <CallToActionLink
          text={intl.formatMessage({
            id: 'health.LifestyleTips.ViewLifestyleTips',
          })}
        />
      </>
    </Container>
  );
};

LifestyleResultsCard.propTypes = {
  color: PropTypes.string.isRequired,
  score: PropTypes.number,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  tips: PropTypes.shape({
    category: PropTypes.string,
    result: PropTypes.string,
    tipDescription: PropTypes.string,
  }),
};

export default LifestyleResultsCard;
