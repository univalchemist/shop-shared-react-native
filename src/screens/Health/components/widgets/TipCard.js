import React from 'react';
import { Text } from '@wrappers/components';
import { useTheme, useIntl } from '@wrappers/core/hooks';
import styled from 'styled-components/native';
import { CallToActionLink } from './CallToActionLink';
import { TouchableHighlight, Linking, View } from 'react-native';
import { externalLinkBlack } from '@images';
import { categories, logAction } from '@store/analytics/trackingActions';

export const TouchableContainer = styled(TouchableHighlight)`
  background-color: ${props => props.theme.colors.white};
  border-radius: 4;
  padding-top: 32;
  margin-bottom: 8;
  shadow-offset: 0px 2px;
  shadow-opacity: 0.05px;
  shadow-radius: 3;
  shadow-color: #000;
  flex-grow: 1;
`;

const Topic = styled(Text)`
  color: ${props => props.theme.colors.gray[0]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  font-weight: ${props => props.theme.fontWeights.bold};
  letter-spacing: 0.3;
  padding-left: 24;
  padding-right: 24;
`;

const Source = styled(Text)`
  color: ${props => props.theme.colors.gray[0]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  letter-spacing: 0.3;
  padding-left: 24;
  padding-right: 24;
`;

const Description = styled(Text)`
  padding-top: 32;
  color: ${props => props.theme.colors.gray[8]};
  font-size: ${props => props.theme.fontSizes[2]};
  line-height: 22;
  letter-spacing: 0.3;
  padding-left: 24;
  padding-right: 24;
  padding-bottom: 32;
`;

export const TipCard = ({ topic, source, text, link }) => {
  const theme = useTheme();
  const intl = useIntl();

  return (
    <TouchableContainer
      accessible={true}
      accessibilityRole={'link'}
      underlayColor={theme.colors.touchableOverlayColor}
      onPress={() => {
        if (link) {
          Linking.openURL(link);
        }

        logAction({
          category: categories.LIFESTYLE_OVERVIEW,
          action: `Learn more ${topic}`,
        });
      }}
      theme={theme}
    >
      <>
        <View flexGrow={1}>
          <Topic accessible={true} accessibilityLabel={topic} flexGrow={1}>
            {topic}
          </Topic>
          <Source accessible={true} accessibilityLabel={source} flexGrow={1}>
            {`${intl.formatMessage({
              id: 'health.LifestyleTips.By',
            })} ${source}`.trim()}
          </Source>
          <Description accessible={true} accessibilityLabel={text} flexGrow={1}>
            {text}
          </Description>
        </View>
        <CallToActionLink
          text={intl.formatMessage({ id: 'health.learnMore' })}
          icon={externalLinkBlack}
        />
      </>
    </TouchableContainer>
  );
};
