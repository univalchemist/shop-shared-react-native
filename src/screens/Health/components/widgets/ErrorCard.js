import React from 'react';
import { TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { CardContainer } from '@screens/Health/components/widgets/CardContainer';
import { useTheme, useIntl } from '@wrappers/core/hooks';
import {
  Box,
  ErrorText,
  SectionHeadingText,
  Button,
} from '@wrappers/components';
import { LIFESTYLE_FORM } from '@routes';

export const ErrorCard = ({ color }) => {
  const navigation = useNavigation();
  const theme = useTheme();
  const intl = useIntl();

  return (
    <TouchableOpacity
      activeOpacity={1}
      onPress={() =>
        navigation.navigate(LIFESTYLE_FORM, { scrollToFutureMe: true })
      }
    >
      <CardContainer
        theme={theme}
        justifyContent="center"
        color={color}
        flexGrow={1}
      >
        <SectionHeadingText textAlign="center">
          {intl.formatMessage({
            id: 'health.healthScoreHistory.errorMessage.somethingWentWrong',
          })}
        </SectionHeadingText>

        <ErrorText
          mt={2}
          textAlign="center"
          accessible={true}
          accessibilityLabel={intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.unableToShow',
          })}
          flexGrow={1}
        >
          {intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.unableToShow',
          })}
        </ErrorText>
        <ErrorText
          textAlign="center"
          accessible={true}
          accessibilityLabel={intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.tryAgain',
          })}
          flexGrow={1}
        >
          {intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.tryAgain',
          })}
        </ErrorText>
        <Box mt={4}>
          <Button
            primary
            title={intl.formatMessage({ id: 'tryAgain' })}
            onPress={() =>
              navigation.navigate(LIFESTYLE_FORM, { scrollToFutureMe: true })
            }
          />
        </Box>
      </CardContainer>
    </TouchableOpacity>
  );
};
