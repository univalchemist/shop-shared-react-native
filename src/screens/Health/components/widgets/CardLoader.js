import React from 'react';
import { TextSkeletonPlaceholder, Box, Divider } from '@wrappers/components';
import { TouchableContainer } from './TipCard';

export const CardLoader = ({ width }) => (
  <TouchableContainer>
    <>
      <Box style={{ paddingLeft: 24, paddingRight: 24 }}>
        <TextSkeletonPlaceholder width={width * 0.8} />
      </Box>
      <Box style={{ paddingTop: 4, paddingLeft: 24, paddingRight: 24 }}>
        <TextSkeletonPlaceholder width={width * 0.5} />
      </Box>
      <Box style={{ paddingTop: 30, paddingLeft: 24, paddingRight: 24 }}>
        <TextSkeletonPlaceholder width={width} />
      </Box>
      <Box style={{ paddingTop: 4, paddingLeft: 24, paddingRight: 24 }}>
        <TextSkeletonPlaceholder width={width} />
      </Box>
      <Box
        style={{
          paddingTop: 4,
          paddingLeft: 24,
          paddingRight: 24,
          paddingBottom: 32,
        }}
      >
        <TextSkeletonPlaceholder width={width * 0.5} />
      </Box>
      <Divider full={true} />
      <Box
        style={{
          paddingTop: 16,
          paddingLeft: 24,
          paddingRight: 24,
          paddingBottom: 16,
        }}
      >
        <TextSkeletonPlaceholder width={width * 0.5} />
      </Box>
    </>
  </TouchableContainer>
);
