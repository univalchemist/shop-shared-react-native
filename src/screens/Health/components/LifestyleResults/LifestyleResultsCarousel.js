import { Pagination } from 'react-native-snap-carousel';
import { TrackedCarousel, Box } from '@wrappers/components';
import LifestyleResultsCard from './LifestyleResultsCard';
import React, { useState } from 'react';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard';
import { useTheme } from '@wrappers/core/hooks';
import { categories } from '@store/analytics/trackingActions';

export const LifestyleResultsCarousel = ({
  data,
  cardWidthWithSpacing,
  horizontalMarginBetweenCards,
  viewportWidth,
  navigation,
}) => {
  const theme = useTheme();
  const [activeIndex, setActiveIndex] = useState(0);

  return (
    <Box marginTop={16}>
      <TrackedCarousel
        inactiveSlideScale={1}
        inactiveSlideOpacity={1}
        data={data}
        onSnapToItem={index => setActiveIndex(index)}
        actionParams={{
          category: categories.LIFESTYLE_OVERVIEW,
          action: 'Horizontal scroll of my results',
        }}
        renderItem={({ item }) => (
          <Box
            width={cardWidthWithSpacing}
            flexGrow={1}
            marginBottom={1}
            px={horizontalMarginBetweenCards / 3}
          >
            {item.isValid ? (
              <LifestyleResultsCard {...item} navigation={navigation} />
            ) : (
              <ErrorCard color={theme.colors.error[3]} />
            )}
          </Box>
        )}
        sliderWidth={viewportWidth}
        itemWidth={cardWidthWithSpacing}
      />
      <Pagination
        containerStyle={{
          paddingVertical: 15,
          paddingBottom: 24,
        }}
        inactiveDotScale={1}
        dotsLength={data.length}
        activeDotIndex={activeIndex}
      />
    </Box>
  );
};
