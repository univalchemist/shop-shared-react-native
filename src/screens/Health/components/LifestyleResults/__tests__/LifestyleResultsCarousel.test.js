import React from 'react';
import { renderForTest } from '@testUtils';
import { LifestyleResultsCarousel } from '../LifestyleResultsCarousel';
import Carousel, { Pagination } from 'react-native-snap-carousel';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard';
import LifestyleResultsCard from '../LifestyleResultsCard';

describe('LifestyleResultsLoader', () => {
  const validResults = {
    score: 20,
    color: 'red',
    status: 'mock status',
    description: 'mock desc',
    title: 'mock title',
    isValid: true,
    textColor: '#333333',
  };

  const invalidResults = {
    score: 20,
    color: 'red',
    status: 'mock status',
    description: 'mock desc',
    title: 'mock title',
    isValid: false,
    textColor: '#333333',
  };

  const renderLifestyleResultsCarousel = data => {
    return renderForTest(
      <LifestyleResultsCarousel
        data={data}
        cardWidthWithSpacing={0}
        cardHeight={250}
        horizontalMarginBetweenCard={3}
        cardWidth={-3}
        viewportWidth={3}
      />,
    );
  };

  it('should match snapshot', () => {
    const component = renderLifestyleResultsCarousel([validResults]);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render the carousel', () => {
    const component = renderLifestyleResultsCarousel([validResults]);
    const carousel = component.queryAllByType(Carousel);

    expect(carousel[0]).toBeDefined();
  });

  it('should render number of cards according to number of data', () => {
    const data = [validResults, invalidResults];
    const component = renderLifestyleResultsCarousel(data);

    const lifestyleResultsCard = component.queryAllByType(LifestyleResultsCard);
    const errorCard = component.queryAllByType(ErrorCard);

    expect(lifestyleResultsCard.length).toBe(1);
    expect(errorCard.length).toBe(1);
  });

  it('should render pagination for carousel', () => {
    const data = [validResults, validResults];
    const component = renderLifestyleResultsCarousel(data);

    const pagination = component.queryAllByType(Pagination);

    expect(pagination.length).toBe(1);
    expect(pagination[0].props.dotsLength).toBe(2);
  });

  it('should render lifestyle results card when isValid passed is true ', () => {
    const data = [validResults];
    const component = renderLifestyleResultsCarousel(data);

    const lifestyleResultsCard = component.queryAllByType(LifestyleResultsCard);

    expect(lifestyleResultsCard.length).toEqual(1);
  });

  it('should render error card when isValid passed is false ', () => {
    const data = [invalidResults];
    const component = renderLifestyleResultsCarousel(data);

    const errorCard = component.queryAllByType(ErrorCard);
    expect(errorCard.length).toEqual(1);
  });
});
