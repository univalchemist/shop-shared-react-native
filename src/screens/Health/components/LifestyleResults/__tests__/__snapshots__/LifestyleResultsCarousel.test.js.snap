// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifestyleResultsLoader should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        marginTop={16}
                        style={
                          Array [
                            Object {
                              "marginTop": 16,
                            },
                          ]
                        }
                      >
                        <RCTScrollView
                          actionParams={
                            Object {
                              "action": "Horizontal scroll of my results",
                              "category": "My lifestyle overview",
                            }
                          }
                          activeAnimationOptions={null}
                          activeAnimationType="timing"
                          activeSlideAlignment="center"
                          activeSlideOffset={20}
                          apparitionDelay={0}
                          automaticallyAdjustContentInsets={false}
                          autoplay={false}
                          autoplayDelay={1000}
                          autoplayInterval={3000}
                          callbackOffsetMargin={5}
                          containerCustomStyle={Object {}}
                          contentContainerCustomStyle={Object {}}
                          contentContainerStyle={
                            Array [
                              Object {
                                "paddingLeft": 1.5,
                                "paddingRight": 1.5,
                              },
                              Object {},
                            ]
                          }
                          data={
                            Array [
                              Object {
                                "color": "red",
                                "description": "mock desc",
                                "isValid": true,
                                "score": 20,
                                "status": "mock status",
                                "textColor": "#333333",
                                "title": "mock title",
                              },
                            ]
                          }
                          decelerationRate="fast"
                          directionalLockEnabled={true}
                          disableVirtualization={false}
                          enableMomentum={false}
                          enableSnap={true}
                          firstItem={0}
                          getItem={[Function]}
                          getItemCount={[Function]}
                          hasParallaxImages={false}
                          horizontal={true}
                          inactiveSlideOpacity={1}
                          inactiveSlideScale={1}
                          inactiveSlideShift={0}
                          initialNumToRender={Infinity}
                          invertStickyHeaders={false}
                          inverted={false}
                          itemWidth={0}
                          keyExtractor={[Function]}
                          layout="default"
                          lockScrollTimeoutDuration={1000}
                          lockScrollWhileSnapping={false}
                          loop={false}
                          loopClonesPerSide={3}
                          maxToRenderPerBatch={Infinity}
                          numColumns={1}
                          onContentSizeChange={[Function]}
                          onEndReachedThreshold={2}
                          onLayout={[Function]}
                          onMomentumScrollEnd={[Function]}
                          onResponderRelease={[Function]}
                          onScroll={[Function]}
                          onScrollBeginDrag={[Function]}
                          onScrollEndDrag={[Function]}
                          onSnapToItem={[Function]}
                          onStartShouldSetResponderCapture={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                          overScrollMode="never"
                          pinchGestureEnabled={false}
                          removeClippedSubviews={true}
                          renderItem={[Function]}
                          scrollEnabled={true}
                          scrollEventThrottle={1}
                          scrollsToTop={false}
                          shouldOptimizeUpdates={true}
                          showsHorizontalScrollIndicator={false}
                          showsVerticalScrollIndicator={false}
                          slideStyle={Object {}}
                          sliderWidth={3}
                          stickyHeaderIndices={Array []}
                          style={
                            Object {
                              "flexDirection": "row",
                              "opacity": 0,
                              "width": 3,
                            }
                          }
                          swipeThreshold={20}
                          updateCellsBatchingPeriod={50}
                          useScrollView={false}
                          vertical={false}
                          viewabilityConfigCallbackPairs={Array []}
                          windowSize={Infinity}
                        >
                          <View>
                            <View
                              onLayout={[Function]}
                              style={
                                Array [
                                  Object {
                                    "flexDirection": "row",
                                  },
                                  null,
                                ]
                              }
                            >
                              <View
                                pointerEvents="box-none"
                                style={
                                  Array [
                                    Object {
                                      "width": 0,
                                    },
                                    Object {},
                                    Object {},
                                  ]
                                }
                              >
                                <View
                                  flexGrow={1}
                                  marginBottom={1}
                                  px={NaN}
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 4,
                                        "paddingLeft": "NaNpx",
                                        "paddingRight": "NaNpx",
                                        "width": "0%",
                                      },
                                    ]
                                  }
                                  width={0}
                                >
                                  <View
                                    accessibilityRole="button"
                                    accessible={true}
                                    focusable={true}
                                    isTVSelectable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "#ffffff",
                                          "borderRadius": 4,
                                          "borderTopColor": "red",
                                          "borderTopWidth": 4,
                                          "display": "flex",
                                          "flexGrow": 1,
                                          "paddingTop": 28,
                                          "shadowColor": "#000",
                                          "shadowOffset": Object {
                                            "height": 2,
                                            "width": 0,
                                          },
                                          "shadowOpacity": 0.05,
                                          "shadowRadius": 3,
                                        },
                                      ]
                                    }
                                  >
                                    <View
                                      flexDirection="row"
                                      flexGrow={1}
                                      px={24}
                                      style={
                                        Array [
                                          Object {
                                            "display": "flex",
                                            "flexDirection": "row",
                                            "flexWrap": "wrap",
                                            "paddingLeft": 24,
                                            "paddingRight": 24,
                                          },
                                        ]
                                      }
                                    >
                                      <View
                                        alignSelf="center"
                                        marginRight={30}
                                        style={
                                          Array [
                                            Object {
                                              "alignSelf": "center",
                                              "marginRight": 30,
                                            },
                                          ]
                                        }
                                      >
                                        <Text
                                          accessibilityLabel="20.0"
                                          accessible={true}
                                          color="#333333"
                                          fontFamily="default"
                                          fontSize={16}
                                          lineHeight={22}
                                          style={
                                            Array [
                                              Object {
                                                "color": "#333333",
                                                "fontFamily": "Arial",
                                                "fontSize": 32,
                                                "fontWeight": "800",
                                                "letterSpacing": -1.5,
                                                "lineHeight": 37,
                                                "marginRight": 30,
                                                "marginTop": 6,
                                              },
                                            ]
                                          }
                                        >
                                          20.0
                                        </Text>
                                      </View>
                                      <View
                                        flexDirection="column"
                                        flexGrow={1}
                                        style={
                                          Array [
                                            Object {
                                              "display": "flex",
                                              "flexDirection": "column",
                                              "flexWrap": "wrap",
                                            },
                                          ]
                                        }
                                      >
                                        <Text
                                          accessibilityLabel="mock title"
                                          accessible={true}
                                          color="gray.1"
                                          flexGrow={1}
                                          fontFamily="default"
                                          fontSize={16}
                                          lineHeight={22}
                                          style={
                                            Array [
                                              Object {
                                                "color": "#212121",
                                                "fontFamily": "Arial",
                                                "fontSize": 16,
                                                "letterSpacing": 0.3,
                                                "lineHeight": 22,
                                              },
                                            ]
                                          }
                                        >
                                          mock title
                                        </Text>
                                        <Text
                                          accessibilityLabel="mock status"
                                          accessible={true}
                                          color="gray.1"
                                          flexGrow={1}
                                          fontFamily="default"
                                          fontSize={16}
                                          lineHeight={22}
                                          style={
                                            Array [
                                              Object {
                                                "color": "#212121",
                                                "fontFamily": "Arial",
                                                "fontSize": 16,
                                                "fontWeight": "700",
                                                "letterSpacing": 0.3,
                                                "lineHeight": 22,
                                              },
                                            ]
                                          }
                                        >
                                          mock status
                                        </Text>
                                      </View>
                                      <Text
                                        accessibilityLabel="mock desc"
                                        accessible={true}
                                        color="gray.1"
                                        flexGrow={1}
                                        fontFamily="default"
                                        fontSize={16}
                                        lineHeight={22}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#666666",
                                              "fontFamily": "Arial",
                                              "fontSize": 16,
                                              "letterSpacing": 0.3,
                                              "lineHeight": 24,
                                              "paddingBottom": 32,
                                              "paddingTop": 32,
                                            },
                                          ]
                                        }
                                      >
                                        mock desc
                                      </Text>
                                    </View>
                                    <View
                                      flexDirection="column"
                                      height={54}
                                      style={
                                        Array [
                                          Object {
                                            "display": "flex",
                                            "flexDirection": "column",
                                            "flexWrap": "wrap",
                                            "height": 54,
                                          },
                                        ]
                                      }
                                    >
                                      <View
                                        backgroundColor="gray.5"
                                        height={1}
                                        style={
                                          Array [
                                            Object {
                                              "backgroundColor": "#e3e3e3",
                                              "height": 1,
                                              "width": "100%",
                                            },
                                          ]
                                        }
                                        width="100%"
                                      />
                                      <View
                                        flexDirection="row"
                                        paddingLeft={24}
                                        paddingRight={24}
                                        paddingTop={16}
                                        style={
                                          Array [
                                            Object {
                                              "display": "flex",
                                              "flexDirection": "row",
                                              "flexWrap": "wrap",
                                              "paddingLeft": 24,
                                              "paddingRight": 24,
                                              "paddingTop": 16,
                                            },
                                          ]
                                        }
                                      >
                                        <Text
                                          color="gray.1"
                                          fontFamily="default"
                                          fontSize={16}
                                          lineHeight={22}
                                          style={
                                            Array [
                                              Object {
                                                "color": "#333333",
                                                "fontFamily": "Arial",
                                                "fontSize": 16,
                                                "lineHeight": 22,
                                              },
                                            ]
                                          }
                                        >
                                          View lifestyle tips
                                        </Text>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
